src/prof.c

```diff
@@ -1705,11 +1710,76 @@ prof_getpid(void) {
 static bool
 prof_dump_maps(bool propagate_err) {
 	bool ret;
-	int mfd;
+	int mfd = -1;
 
 	cassert(config_prof);
 #ifdef __FreeBSD__
-	mfd = prof_open_maps("/proc/curproc/map");
+	int mib[4];
+	size_t sz;
+	char *buf;
+
+	mib[0] = CTL_KERN;
+	mib[1] = KERN_PROC;
+	mib[2] = KERN_PROC_VMMAP;
+	mib[3] = prof_getpid();
+
+	if (prof_dump_write(propagate_err, "\nMAPPED_LIBRARIES:\n") &&
```

Comment:
The FreeBSD test is failing here https://master.jemalloc.net/#/builders/13/builds/383/steps/4/logs/stdio. @devnexen can you look into it?

Reply:
Is --enable-prof passed to configure ? Could the system jemalloc possibly interferes too (eg invalid pair) ?

Reply:
Yes that test is with config prof. I did gdb the test, and it was failing on this branch. The invalid conf pair warnings in the log is from the system jemalloc, not the one being tested.

Reply:
Odd (can t access freebsd H/W for now) that s a basic string, so normally I should not be the assertion failing if this enabled with profiling ...

Reply:
This is what I get in my side
`

...
=== test/unit/prof_accum ===
<jemalloc>: Invalid conf pair: prof:true
<jemalloc>: Invalid conf pair: prof_accum:true
<jemalloc>: Invalid conf pair: prof_active:false
<jemalloc>: Invalid conf pair: lg_prof_sample:0
test_idump (non-reentrant): pass
--- pass: 1/1, skip: 0/1, fail: 0/1 ---
...
=== test/unit/prof_reset ===
<jemalloc>: Invalid conf pair: prof:true
<jemalloc>: Invalid conf pair: prof_active:false
<jemalloc>: Invalid conf pair: lg_prof_sample:0
test_prof_reset_basic (non-reentrant): pass
test_prof_reset_cleanup (non-reentrant): pass
test_prof_reset (non-reentrant): pass
test_xallocx (non-reentrant): pass
--- pass: 4/4, skip: 0/4, fail: 0/4 ---
`

config.log
`
This file contains any messages produced by compilers while
running configure, to aid debugging if configure makes a mistake.

It was created by configure, which was
generated by GNU Autoconf 2.69.  Invocation command line was

  $ ./configure --enable-prof --enable-debug --enable-log

## --------- ##
## Platform. ##
## --------- ##

hostname = excalibur
uname -m = amd64
uname -r = 13.0-CURRENT
uname -s = FreeBSD
uname -v = FreeBSD 13.0-CURRENT b4ed6590307(master) GENERIC 
`